truffle/src/main/ruby/core/array.rb

```diff
@@ -359,12 +362,12 @@ def fill_internal(a=undefined, b=undefined, c=undefined)
       right += 1 unless one.exclude_end?
       return self if right <= left           # Nothing to modify
 
-    elsif one and !undefined.equal?(one)
+    elsif one
       left = Rubinius::Type.coerce_to_collection_length one
       left += size if left < 0
       left = 0 if left < 0
 
-      if two and !undefined.equal?(two)
+      if !undefined.equal?(two) and two
```

Comment:
Just my opinion: second one looks like Perl and `and` is needed for assignment in conditionals.